map/src/main/java/openlr/map/utils/GeometryUtils.java

```diff
@@ -316,50 +316,23 @@ private static double hMult(final double y) {
     public static double calculateLineBearing(final Line line,
                                               final BearingDirection dir, final int pointDistance,
                                               final int projectionAlongLine) {
-        if (line == null) {
+        if (line == null || projectionAlongLine < 0 || projectionAlongLine > line.getLineLength()) {
             return -1.0;
         }
         GeoCoordinates p1 = null;
         GeoCoordinates p2 = null;
 
+        BearingDestinationRouting bearingDestinationRouting = BearingDestinationRouting.withConfig(line, pointDistance, projectionAlongLine);

```

Comment:
I wonder about creating a new object for each bearing calculation.  Perhaps static utility methods would be the better choice.  Anyone else have an opinion on this?

Reply:
482ec5f58091993383b0ec513ceafb5223fa03d7